pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
 	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
 
+	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
+		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
```

Comment:
You mean a new config flag for setting docker version and then setting serializeimagepull based on that new flag? Somehow I think it could be more error prone even for admins, 1) what if the new flag and serializeimagepulls are configured incorrectly and not in accordance to each other, 2) what if docker is upgraded or downgraded and an admin forgets to reconfigure any of these flag.

I was thinking more about it and I am still wondering why is it an issue to determine docker version at run time (infact to avoid manual errors) and then setting config vars accordingly? I am still trying to understand if  there is any technical reason for that?